modules/lightning_telemetry/src/Telemetry.php

```diff
@@ -0,0 +1,188 @@
+   * Sends an event to Amplitude.
+   *
+   * @param \Drupal\lightning_telemetry\Event $event
+   *   The Amplitude event.
+   *
+   * @return bool
+   *   TRUE if the request to Amplitude was successful.
+   */
+  protected function sendEvent(Event $event) {
+    $response = $this->httpClient->request('POST', self::AMPLITUDE_API_URL, [
+      'form_params' => [
+        'api_key' => self::AMPLITUDE_API_KEY,
+        'event' => $event->__toJson(),
+      ]
+    ]);
+    $success = $response->getStatusCode() == 200;
+
+    return $success;
+  }
+
+  /**
+   * Creates and sends a cron event to Amplitude.
+   *
+   * @param string $event_type
+   *   The event type.
+   *
+   * @param array $event_properties
+   *   Event properties.
+   *
+   * @throws \Exception
+   */
+  public function sendTelemetry($event_type, array $event_properties = []) {
+    $event = $this->createEvent($event_type, $event_properties);
+
+    // Failure to send Telemetry should never cause a user facing error or
+    // interrupt a process. Telemetry failure should be graceful and quiet.
+    try {
+      $this->sendEvent($event);
+    }
+    catch (\Exception $e) {
+      if (getenv('LIGHTNING_TELEMETRY_LOUD')) {
+        throw $e;
+      }
+    }
+  }
+
+  /**
+   * Get an array of information about Lightning extensions.
+   *
+   * @return array
+   *   An array of extension info keyed by the extensions machine name.
+   */
+  protected function getExtensionInfo() {
+    $all_modules = $this->extensionListModule->getAllAvailableInfo();
+    $lightning_extensions = array_intersect_key($all_modules, array_flip($this->getLightningExtensionNames()));
+    $extension_info = [];
+
+    foreach ($lightning_extensions as $name => $extension) {
+      // Version is unset for dev versions.
+      $version = $extension['version'] ? $extension['version'] : $extension['core'];
+      $extension_info[$name]['version'] = $version;
+    }
+
+    $installed_modules = $this->extensionListModule->getAllInstalledInfo();
+    foreach ($lightning_extensions as $name => $extension) {
+      if (array_key_exists($name, $installed_modules)) {
+        $extension_info[$name]['status'] = 'enabled';
+      }
+      else {
+        $extension_info[$name]['status']  = 'disabled';
+      }
+    }
+
+    return $extension_info;
+  }
+
+  /**
+   * Creates an Amplitude event.
+   *
+   * @param string $type
+   *   The event type.
+   * @param array $properties
+   *   The event properties.
+   *
+   * @return \Drupal\lightning_telemetry\Event
+   *   An Amplitude event with basic info already populated.
+   */
+  protected function createEvent($type, $properties) {
+    $user_id = $this->config_factory->get('system.site')->get('uuid');
+    $default_properties['extensions'] = $this->getExtensionInfo();
+    $default_properties['php']['version'] = phpversion();
+    $default_properties['drupal']['version'] = \Drupal::VERSION;
+    $properties = array_merge_recursive($default_properties, $properties);
```

Comment:
I think we should probably use Drupal\Component\Utility\NestedArray::mergeDeep() here.